src/System.Device.Gpio/System/Device/Gpio/Drivers/RaspberryPi3Driver.Linux.cs

```diff
@@ -103,11 +85,19 @@ protected internal override void ClosePin(int pinNumber)
         {
             ValidatePinNumber(pinNumber);
 
-            if (_sysFSModes.ContainsKey(pinNumber) && _sysFSModes[pinNumber] == PinMode.Output)
+            lock (s_sysFsInitializationLock)
```

Comment:
NIT: if we make this dictionary a concurrentdictionary then I believe we won't need this lock any longer.

Reply:
The lock here is wrong altogether, since it's for the driver, not the _pinModes array. I'll just init the driver in the ctor and forget about locking the driver.